### DISCUSSION TITLE
Advice on testing aspect-introduced types

### DISCUSSION THREAD

**Discussion by WhitWaldo** [metalama/Metalama#353] [Q&A] [answered]
As with all other aspects, I'd like to create aspect tests to validate that the aspects are generating as expected. Typically, after applying the `Metalama.Testing.AspectTesting` library, I'd go about generating the .t.cs files for each of the targeted files in the project. But when the class is being introduced, there's so such file to start with, so how might you advise going about testing this?

Thanks!

**Comment by gfraiteur** [comment, marked as answer]
For new top-level types, the test framework generates a file named `.i.cs`. The approach is to let the test framework generate this file and, when you are satisfied with the result, accept these files in your project. Using `DiffEngineTray` helps in this process.

**Comment by WhitWaldo** [comment]
Excellent, I see it now. Thank you!
